workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/StandardGameObjectCreation/GameObjectCreatorFromMetadata.cs

```diff
@@ -0,0 +1,82 @@
+using System.Collections.Generic;
+using System.IO;
+using Generated.Improbable;
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Core.GameObjectRepresentation;
+using UnityEngine;
+using Transform = Generated.Improbable.Transform.Transform;
+
+namespace Improbable.Gdk.Core.GameObjectRepresentation
+{
+    public class GameObjectCreatorFromMetadata : IEntityGameObjectCreator
+    {
+        private readonly Dictionary<string, GameObject> cachedPrefabs
+            = new Dictionary<string, GameObject>();
+
+        private readonly string workerType;
+
+        public GameObjectCreatorFromMetadata(string workerType)
+        {
+            this.workerType = workerType;
+        }
+
+        public GameObject GetGameObjectForEntityAdded(SpatialOSEntity entity, WorkerSystem worker)
+        {
+            if (!entity.HasComponent<Metadata.Component>())
+            {
+                return null;
```

Comment:
Would it make sense to add a method to `IEntityGameObjectCreator` for something like `bool NewlyAddedEntityShouldBeRepresentedByAGameObject`? Alternatively, shall we let `GetGameObjectForEntityAdded()` return a bool? Returning a null GameObject to tell users that an entity doesn't need a GameObject feels like a gotcha.